server/services/curiousService.ts

```diff
@@ -0,0 +1,31 @@
+import type { LearnerNeurodivergence, LearnerProfile } from 'curiousApiClient'
+import type { SupportNeeds } from 'viewModels'
+import { toSupportNeeds } from '../routes/overview/mappers/supportNeedsMapper'
+import CuriousClient from '../data/curiousClient'
+import { HmppsAuthClient } from '../data'
+
+export default class CuriousService {
+  constructor(private readonly hmppsAuthClient: HmppsAuthClient, private readonly curiousClient: CuriousClient) {}
+
+  async getPrisonerSupportNeeds(
+    prisonNumber: string,
+    establishmentId: string,
+    username: string,
+  ): Promise<SupportNeeds> {
+    // TODO - is this the correct token?
+    const systemToken = await this.hmppsAuthClient.getSystemClientToken(username)
```

Comment:
lol - yes, that is the correct token.
We want to use the system token in this case (and also when searching/retrieving prisoners in prisoner-search). This is because both APIs require a role(s), and we dont want the overheard of having to ensure all our users (the CIAGs etc) have these roles. It is OK to ensure that CIAGs have the role for _our_ service UI and API (which we do already), but to say that our users need roles from other services is an overhead and a security risk (they might get access to more than our service allows by directly hitting the other API; or other services they use might suddenly give them access to stuff they are not entitled to in context of that app)

So, the general pattern I'm led to believe is that the UI service itself calls these other APIs, using the system token.
As you can see in what you've done here (assume copied from prisoner-search) is that we also pass in the logged in username when getting the system token. This is quite neat - what it does is get the system token, but with a "modified" username claim in the JWT, representing the authenticated user. This means that logs etc in the other system will detail the logged in username, so gives some more context to the logs etc of those other systems.

Reply:
I thought as much - I knew you said we didn't want the overhead of ensuring each user has specific roles for other systems. I just wasn't sure if this was the correct "system" token :)